### PR TITLE
Web3 method filter fix

### PR DESCRIPTION
**Pull request by diega**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 1 issue

1. ![MINOR][MINOR] [JsonRpcWeb3ServerHandler.java#L53](https://github.com/rsksmart/rskj/blob/b25decc5472784f88465873eefbca98ac562f334/rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3ServerHandler.java#L53): Make this final field static too. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1170)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/rpc/JsonRpcMethodFilter.java [hunk 1]
```diff
@@ -19,47 +19,35 @@
...
-        super(new ObjectMapper(), handler, remoteInterface);
-
+    public JsonRpcMethodFilter(List<ModuleDescription> modules) {
         this.modules = modules;
     }
 
     @Override
-    protected ErrorResolver.JsonError handleJsonNodeRequest(JsonNode node, OutputStream output) throws IOException {
+    public void interceptRequest(JsonNode node) throws IOException {
         if (node.hasNonNull(JSON_RPC_METHOD_FIELD_NAME)) {
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/rpc/JsonRpcMethodFilter.java]: Maybe we could require that `method` is always present.

Comment (diega) [reply rskj-core/src/main/java/co/rsk/rpc/JsonRpcMethodFilter.java]: This is already checked in the `com.googlecode.jsonrpc4j.JsonRpcBasicServer#isValidRequest` method invoked by _handleObject_

**Comment by rskops** [comment]
SonarQube analysis reported 1 issue

1. ![MINOR][MINOR] [JsonRpcWeb3ServerHandler.java#L53](https://github.com/rsksmart/rskj/blob/08744277b51c9e9038b2f88f4e96a98bbf253dc1/rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3ServerHandler.java#L53): Make this final field static too. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1170)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 1 issue

1. ![MINOR][MINOR] [JsonRpcWeb3ServerHandler.java#L53](https://github.com/rsksmart/rskj/blob/c303642e93bead9fe96bd19b456b0a17926629e9/rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3ServerHandler.java#L53): Make this final field static too. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1170)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 1 issue

1. ![MINOR][MINOR] [JsonRpcWeb3ServerHandler.java#L53](https://github.com/rsksmart/rskj/blob/eefd3c1b7aa0ee9ec9fb41b43547f614754dd312/rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3ServerHandler.java#L53): Make this final field static too. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1170)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**